pkg/build/registry/build/strategy.go

```diff
@@ -109,27 +110,51 @@ type detailsStrategy struct {
 }
 
 // Prepares a build for update by only allowing an update to build details.
-// For now, this is the Spec.Revision field
+// Build details currently consists of: Spec.Revision, Status.Reason, and
+// Status.Message, all of which are updated from within the build pod
 func (detailsStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
 	revision := newBuild.Spec.Revision
+	message := newBuild.Status.Message
+	reason := newBuild.Status.Reason
 	*newBuild = *oldBuild
 	newBuild.Spec.Revision = revision
+	newBuild.Status.Reason = reason
+	newBuild.Status.Message = message
 }
 
 // Validates that an update is valid by ensuring that no Revision exists and that it's not getting updated to blank
 func (detailsStrategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) field.ErrorList {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
+	oldRevision := oldBuild.Spec.Revision
+	newRevision := newBuild.Spec.Revision
+	oldReason := oldBuild.Status.Reason
+	oldMessage := oldBuild.Status.Message
+	newReason := newBuild.Status.Reason
+	newMessage := newBuild.Status.Message
 	errors := field.ErrorList{}
-	if oldBuild.Spec.Revision != nil {
```

Comment:
That way we preserve the immutability of the build spec. We don't allow other changes to the spec, and setting the build revision is simply completing that spec.

And now that I think about it, it doesn't make sense to allow you to update the spec and status in the same /details endpoint. But that may be too late.